### DISCUSSION TITLE
Onboarding shows Pin dependencies despite setting rangeStrategy

### DISCUSSION THREAD

**Discussion by mfinegold-cambia** [renovatebot/renovate#9751] [Request Help]
My package.json uses semver ranges ( e.g. ^1.0.1 ) and I'd like Renovate to continue to use a range when creating update MR's ( ^1.0.1 -> ^1.2.0), rather than pin to a version.  But when I set rangeStrategy to 'replace' or 'bump', the Onboarding MR page in Gitlab still shows that it will create a "Pin Dependencies" MR and the other MR's show that it will upgrade to a pinned version rather than semver range.  My config.js is below - is there something I'm missing here?

module.exports = {
  platform: 'gitlab',
  endpoint: 'my_gitlab_api_endpoint',
  repositories: [ process.env.CI_PROJECT_PATH ],
  rangeStrategy: 'bump',
  branchPrefix: 'dependency-update/',
  automerge: false,
  onboarding: true,
  onboardingConfig: { "$schema": "https://docs.renovatebot.com/renovate-schema.json", "extends": ["config:base"] },
  onboardingPrTitle: 'RenovateTest47',
  prConcurrentLimit: 10,
  logFileLevel: 'debug',
  major: {
    enabled: true,
  },
  minor: {
    enabled: true
  },
  patch: {
    enabled: true
  },
  packageRules: [
    {
      matchUpdateTypes: [ "patch", "minor"],
      groupName: "Minor and Patch Versions",
    },
    {
      matchUpdateTypes: [ "major"],
      matchDepTypes: [ "devDependencies"],
      groupName: "devDependencies"
    },
    {
      matchPackagePrefixes: [ "@chp-cps" ],
      enabled: false
    }
  ],
  includePaths: [
    "package.json",
     "**cucumber-mock/package.json"
   ],
  ignorePaths: [
    "**/node_modules/**"
  ],
  vulnerabilityAlerts: {
    enabled: true,
    labels: [
       "vulnerability",
     ],
   },
  includeForks: true,
  printConfig: true,
  labels: ['dependency-scanning']
};

**Comment by rarkins** [comment]
Because config:base in your onboarding config overrides rangeStrategy.

Either:
- move rangeStrategy=bump into your onboardingConfig, or
- move all this type of repo config into your own preset and extend that in onboardingConfig instead of config:base

**Reply by mfinegold-cambia** [reply]
Thanks so much - that helps.  A few other questions:
 
- I'm trying to group patch and minor versions ( "Minor and Patch Versions") into 1 MR in Gitlab.  My packageRules are below.  The Minor version upgrades appear in the group, but renovate creates another MR/branch for a lodash upgrade from ^4.17.20 -> ^4.17.21 .   I would expect that patch update to appear in my group?  Oddly, a patch update for moment-timezone does appear in my group - ^0.5.23 -> ^0.5.33

  packageRules: [
    {
      matchUpdateTypes: [ "minor", "patch" ],
      groupName: "Minor and Patch Versions"
    },
    {
      matchUpdateTypes: [ "major"],
      matchDepTypes: [ "devDependencies"],
      groupName: "devDependencies"
    }
  ]

- The docs for vulnerabilityAlerts reference some functionality in Github, but does this work at all with Gitlab?  If so, any info on how to set that up?

**Reply by rarkins** [reply]
If you have configured separateMinorPatch then that would explain it the grouping you see.

vulnerability alerts come from GitHub so only apply to GitHub.

**Reply by tennox** [reply]
I tried to override the setting both ways (`extends` and `rangeStrategy` directly), but still get Pin dependencies MRs:
```json5
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:js-app",
    "group:all",
    ":preserveSemverRanges" // disable automatic pinning - https://docs.renovatebot.com/dependency-pinning/
  ],
  "rangeStrategy": "replace",
  "lockFileMaintenance": {
    "enabled": true
  }
}
```

Did I misunderstand something?

**Reply by ishowta** [reply]
> I tried to override the setting both ways (`extends` and `rangeStrategy` directly), but still get Pin dependencies MRs:
> 
> ```js
> {
>   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
>   "extends": [
>     "config:js-app",
>     "group:all",
>     ":preserveSemverRanges" // disable automatic pinning - https://docs.renovatebot.com/dependency-pinning/
>   ],
>   "rangeStrategy": "replace",
>   "lockFileMaintenance": {
>     "enabled": true
>   }
> }
> ```
> 
> Did I misunderstand something?

Maybe this one. https://github.com/renovatebot/renovate/issues/13111
